### ISSUE TITLE
Button on Narrative Templates

### ISSUE THREAD

**Issue by ppf-mcmasty** [sendwithus/templates#25]
Hi,

I am less than a novice at CSS... was hoping to grab & hack these templates to success for my usage.

I am trying to use the narrative templates and when sent to Google, etc it seems to work OK, but some of our customers are are not get an active button.  Specifically the confirmation email.

We also use InfusionSoft, so when I just attached this email to the contact history as a 'sent mail', without actually sending the email body, the HTML <a> tag is getting stripped out.   I suspect that InfusionSoft is using some sort of Outlook engine under the hood.

When I remove the <a> tag from wrapping an inner <table> it seems to work, but it does not look at nice.

It also looks like the `urn:schemas-microsoft-com:office:word` has been deprecated.

https://msdn.microsoft.com/en-us/library/ms875215(v=exchg.65).aspx

Since the "button" is contained by all the MSO stuff, I suspect his is the root cause.  

Any Insights ?

Thanks in advance.

Tyler

**Comment by ppf-mcmasty** [comment]
FWIW,

I have locally replaced lines 242 - 255 with something like the follow

``` 
                    <tr>
                      <td class="button">
                        <div><!--[if mso]>
                          <v:roundrect xmlns:v="urn:schemas-microsoft-com:vml" xmlns:w="urn:schemas-microsoft-com:office:word" href="http://" style="height:45px;v-text-anchor:middle;width:155px;" arcsize="0%" strokecolor="#ffffff" fillcolor="##D84A38">
                            <w:anchorlock/>
                            <center style="color:#ffffff;font-family:Helvetica, Arial, sans-serif;font-size:14px;font-weight:regular;">Submit</center>
                          </v:roundrect>
                        <![endif]--><a href="http://"
                        style="background-color:#D84A38;border-radius:0px;color:#ffffff;display:inline-block;font-family:'Cabin', Helvetica, Arial, sans-serif;font-size:14px;font-weight:bold;line-height:40px;text-align:center;text-decoration:none;width:155px;-webkit-text-size-adjust:none;mso-hide:all;">Submit</a></div>
                      </td>
                    </tr>

 

```  

It seems to have done the trick.

**Comment by FlipCodes** [comment]
Hey @ppf-mcmasty ,

Glad to see you have found something that works. I suspect InfusionSoft is doing something strange to affect this. The deprecated tags shouldn't effect the HTML comments.

Quick question, did you remember to inline the email's CSS? https://github.com/sendwithus/templates#use-with-css-inliner

**Comment by ppf-mcmasty** [comment]
Hi @FlipCodes 

Yup.  I tried Litmus / Putsmail.  It had no impact on the button functionality. Which was it was not an action text / link.

Basically no combination would work so long as the inner table was inside the <a> tag.

Obviously, when I found the snippet above, it removed the whole thing (but also does not have have an 
```
<a href="http://"><table><tr>....</tr></table> </a> 
```

type structure of the original template.

I borrowed those lines of code from the most recent sendwithus template...and replaced the whole `<tr>`

**Comment by FlipCodes** [comment]
Hey @ppf-mcmasty ,

We run all our templates through Litmus and it seems to work on all email clients on our end. Take a look here: https://litmus.com/pub/0336c44/screenshots

It sounds like an InfusionSoft doing something… Glad to know your fix worked!

**Comment by ppf-mcmasty** [comment]
hi @FlipCodes,

I don't think what I am trying to say, is getting across.

1.  It is not the layout that is broken.  It visually looks as expected in all situations.  it is the _link (button)_ that is broken.

2.  We sent this email via MailGun and our customer complained the link didn't work. InfusionSoft is not part of the sending flow at all.  

3.  I also tested this in Litmus.  The view you shared, does not allow us to test the links (we are on the trial version).  The issue is not a layout / presentation issue, but the `<a>` tag being stripped out.  (Even after CSS inlining)

4.  For audit / history purposes, We attach a copy of the email body to the Contact record in InfusionSoft.    By looking at the email source code of the various emails as it was being attached to InfusionSoft the `<a>` tag was getting stripped out.  My hypothesis was the "inner" table, inside the `<a>` html tag of the template.  (Which matched the behavior of the customer.)   


5.  I don't have Outlook, so the only consistent place I could test was to look at the email that stored as part of the Contact History in InfusionSoft.  Including my use of InfusionSoft as a second view of the HTML source, probably ended up being a red herring and only confused the issue for you.

6.  The only way I got the link to remain in the email, was to remove the lines mentioned above.  That is, by making the change above, the `<a>` tag  is now present on the email in the Contact History in InfusionSoft.   I strongly believe there is bug in the template.


***  
Issue Recreation steps.

I just downloaded from your website the latest copy of the narrative template.  https://www.sendwithus.com/resources/templates/narrative  

The Confirm template.


I created real links.  Specifically, I modified lines 245 & 250 replacing `#` with `https://www.sendwithus.com`   No other changes have been made.  (attaching for evidence)

I viewed the HTML in a web-browser to confirm the button actually works.  It does.  Clicking on the "My Order" button takes you to sendwithus website.

Here is my modified HTML body, saved as txt file so that I can attach to GitHub.  Just change the extension back to html, and you should be able to view it a web-browser, send as HTML email, etc.

[send-with-us-raw.txt](https://github.com/sendwithus/templates/files/120428/send-with-us-raw.txt)

I think tested this email with puts mail.

https://putsmail.com/tests/new

Selecting the CSS Inline Option.

Attached below is the raw source, mime-part of the email that was delivered.

If you look at lines 207 - 238 of email source,  you will see that there is no `<a>` tag anywhere in that `<tr>` .  So it is impossible for the for the link to `https://www.sendwithus.com` to work.

[raw-email.txt](https://github.com/sendwithus/templates/files/120435/raw-email.txt)


In a nutshell, I don't think the link works in the template as it is currently coded.   Can you verify this finding ?  

If you repeat the steps with puts mail, I just documented it definitely will strip out the `<a>` tag.   But I don't know if the Litmus link tester will also spot this issue.

Thanks in advance.

**Comment by FlipCodes** [comment]
Hey @ppf-mcmasty ,

Sorry for the delayed response. Thanks for the detailed report, it definitely helped us and I can confirm that the link is broken in the template.  

The issue stems from the link which had a table element in it. The block element(table) is within an inline element(a) and is invalid HTML. That's why it breaks in some email clients.

We will fix and update that template. Thanks for catching that! 

Can you send us an email at support at sendwithus dot com?
